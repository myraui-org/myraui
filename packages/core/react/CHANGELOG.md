# 2.0.0 (2024-10-04)


### üöÄ Features

- **components:** create toast component and update dependencies ([f7b56fd](https://github.com/myraui/myraui/commit/f7b56fd))

- ‚ö†Ô∏è  **components:** use single component folder ([4404384](https://github.com/myraui/myraui/commit/4404384))

- **theme:** set up myra ui theme plugin ([cc5e4f0](https://github.com/myraui/myraui/commit/cc5e4f0))


### ‚ö†Ô∏è  Breaking Changes

- ‚ö†Ô∏è  **components:** use single component folder ([4404384](https://github.com/myraui/myraui/commit/4404384))

### ‚ù§Ô∏è  Thank You

- Moses Gitau @gitaumoses4

## 1.5.2 (2024-09-30)

This was a version bump only for @myraui/react to align it with other projects, there were no code changes.

## 1.5.1 (2024-09-30)


### ü©π Fixes

- **build:** fix pack process ([8397462](https://github.com/myraui/myraui/commit/8397462))


### ‚ù§Ô∏è  Thank You

- Moses Gitau @gitaumoses4

## 1.6.0 (2024-09-30)

This was a version bump only for @myraui/react to align it with other projects, there were no code changes.

## 1.5.0 (2024-09-30)

This was a version bump only for @myraui/react to align it with other projects, there were no code changes.

## 1.4.0 (2024-09-30)

This was a version bump only for @myraui/react to align it with other projects, there were no code changes.