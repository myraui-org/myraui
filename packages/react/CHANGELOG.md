## 0.1.1 (2024-10-28)


### 🩹 Fixes

- **plugin:** add plugin export ([00bdda5](https://github.com/myraui-org/myraui/commit/00bdda5))


### ❤️  Thank You

- Moses Gitau @gitaumoses4

## 0.1.0 (2024-10-28)


### 🚀 Features

- **packages:** remove components package ([dabc145](https://github.com/myraui-org/myraui/commit/dabc145))

- **components:** restructure component themes ([d3b0fcc](https://github.com/myraui-org/myraui/commit/d3b0fcc))

- **components:** update toast theming style ([39ad391](https://github.com/myraui-org/myraui/commit/39ad391))

- **components:** create theme toggler and remove theme selector ([635d04b](https://github.com/myraui-org/myraui/commit/635d04b))

- **components:** update theme icon to be non-filled ([2ac3c13](https://github.com/myraui-org/myraui/commit/2ac3c13))


### 🩹 Fixes

- update external deps for building ([26f999b](https://github.com/myraui-org/myraui/commit/26f999b))

- **packages:** update package json imports ([090956e](https://github.com/myraui-org/myraui/commit/090956e))

- **packages:** update module type ([479eefa](https://github.com/myraui-org/myraui/commit/479eefa))

- **vite:** create base vite config ([2ede68c](https://github.com/myraui-org/myraui/commit/2ede68c))


### ❤️  Thank You

- Moses Gitau @gitaumoses4
