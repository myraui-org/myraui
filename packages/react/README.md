# @myraui/react

A Quick description of the package

## Installation

```sh
yarn add @myraui/react
# or
npm i @myraui/react
```

## Licence

This project is licensed under the terms of the
[MIT license](https://github.com/myraui/myraui/blob/main/LICENSE).
