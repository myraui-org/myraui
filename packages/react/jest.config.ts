/* eslint-disable */
export default {
  displayName: '@myraui-org/react',
  preset: '../../jest.preset.js',
  coverageDirectory: '../../coverage/packages/react',
};
