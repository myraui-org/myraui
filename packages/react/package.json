{
  "name": "@myraui/react",
  "version": "2.2.5",
  "type": "commonjs",
  "description": "A modern React UI library to elevate your UI experience",
  "keywords": [
    "react"
  ],
  "author": {
    "name": "Moses Gitau",
    "email": "gitaumoses4@gmail.com"
  },
  "license": "MIT",
  "main": "./src/index.ts",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/myraui/myraui.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/myraui/myraui/issues"
  },
  "scripts": {
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "peerDependencies": {
    "@nextui-org/react": ">=2.4.8",
    "@tabler/icons-react": ">=3.17.0",
    "react": ">=18",
    "react-dom": ">=18"
  },
  "devDependencies": {
    "@myraui/react-test-utils": "2.2.5",
    "@myraui/test-utils": "2.2.5",
    "@nextui-org/react": "^2.4.8",
    "@tabler/icons-react": "^3.17.0",
    "clean-package": "2.2.0"
  },
  "dependencies": {
    "@myraui/components": "2.2.5",
    "@myraui/hooks": "2.2.5",
    "@myraui/react-utils": "2.2.5",
    "@myraui/shared-utils": "2.2.5",
    "@myraui/theme": "2.2.5"
  },
  "clean-package": "../../clean-package.config.json"
}
