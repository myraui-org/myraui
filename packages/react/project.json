{
  "name": "@myraui-org/react",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/react/src",
  "projectType": "library",
  "tags": ["scope:app"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "build": {
      "executor": "@nx/vite:build",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "packages/react/dist"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/react/jest.config.ts"
      }
    }
  }
}
