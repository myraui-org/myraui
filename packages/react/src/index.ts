'use client';
export * from './components';
export * from '@myraui-org/react-utils';
export * from '@myraui-org/shared-utils';
