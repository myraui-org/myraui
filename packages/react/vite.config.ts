import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import dts from 'vite-plugin-dts';
import * as path from 'path';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import packageJson from './package.json';

export default defineConfig({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/packages/react',

  plugins: [
    react(),
    nxViteTsPaths(),
    dts({
      entryRoot: 'src',
      tsconfigPath: path.join(__dirname, './tsconfig.lib.json'),
    }),
  ],
  build: {
    outDir: './dist',
    reportCompressedSize: true,
    commonjsOptions: {
      transformMixedEsModules: true,
    },
    lib: {
      entry: 'src/index.ts',
      name: '@myraui/react',
      fileName: 'index',
      formats: ['es', 'cjs'],
    },
    rollupOptions: {
      external: [
        ...Object.keys(packageJson.dependencies || {}),
        ...Object.keys(packageJson.peerDependencies || {}),
        ...Object.keys(packageJson.devDependencies || {}),
      ],
    },
  },
});
