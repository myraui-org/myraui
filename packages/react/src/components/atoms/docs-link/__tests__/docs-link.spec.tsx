import * as React from 'react';
import { render } from '@testing-library/react';
import DocsLink from '../docs-link';

describe('DocsLink', () => {
  it('should render correctly', () => {
    const wrapper = render(<DocsLink href="#title">Title</DocsLink>);

    expect(wrapper.container).toMatchSnapshot();

    expect(() => wrapper.unmount()).not.toThrow();
  });
});
