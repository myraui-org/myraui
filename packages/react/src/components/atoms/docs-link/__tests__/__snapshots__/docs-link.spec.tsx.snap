// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsLink should render correctly 1`] = `
<div>
  <a
    class="tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity group relative flex items-center gap-2"
    href="#title"
    role="link"
    tabindex="0"
  >
    Title
    <span
      class="text-primary dark:text-primary-300 opacity-0 transition-opacity group-hover:opacity-100"
    >
      #
    </span>
  </a>
</div>
`;
